pkg/azureclients/armclient/azure_armclient.go

```diff
@@ -95,17 +122,23 @@ func New(authorizer autorest.Authorizer, clientConfig azureclients.ClientConfig,
 		apiVersion:       apiVersion,
 		regionalEndpoint: fmt.Sprintf("%s.%s", clientConfig.Location, url.Host),
 	}
+	client.client.Sender = autorest.DecorateSender(client.client.Sender, sendDecoraters...)
 	client.client.Sender = autorest.DecorateSender(client.client,
```

Comment:
We lose the passed in decorators here and restart the chain with `client.client`. Shouldn't this be `client.client.Sender` to preserve the change? Is the chaining order significant here?